src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorProjectEngine.cs

```diff
@@ -217,6 +217,7 @@ private static void AddComponentFeatures(RazorProjectEngineBuilder builder)
             ComponentLayoutDirective.Register(builder);
             ComponentPageDirective.Register(builder);
             ComponentTypeParamDirective.Register(builder);
+            ComponentPreserveWhitespaceDirective.Register(builder);
```

Comment:
We might have to introduce a new LangVersion and do this only for RazorLangVersion = 5.0 or newer. We don't want this directive to start working for 3.0 projects.

Reply:
Yup need to introduce a new lang version for this. The biggest concern is without a lang version if you use a "new" IDE aka 16.8 VS without a new lang version it'll offer this directive in completions and wont complain when you use it in your files. However, the second you build this will cause build-time errors because it'll be using a different SDK.

Reply:
OK, sounds good. Thanks for the explanation!

@pranavkm has volunteered to help with defining the new lang version, since apparently there are some subtleties has has context on related to controlling which Razor extension gets used inside VS.

Hopefully we'll have this in time to all go in at once in time for preview 7 (but if not, we could define the razorlangversion separately afterwards).

Reply:
Done